tests/kafkatest/services/performance/producer_performance.py

```diff
@@ -13,21 +13,45 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from ducktape.utils.util import wait_until
+
 from kafkatest.services.monitor.jmx import JmxMixin
 from kafkatest.services.performance import PerformanceService
-import itertools
 from kafkatest.services.security.security_config import SecurityConfig
-from kafkatest.services.kafka.directory import kafka_dir
+from kafkatest.services.kafka.directory import kafka_dir, KAFKA_TRUNK
+from kafkatest.services.kafka.version import TRUNK, LATEST_0_8_2
+
+import os
+import subprocess
+
 
 class ProducerPerformanceService(JmxMixin, PerformanceService):
 
+    PERSISTENT_ROOT = "/mnt/producer_performance"
+    STDOUT_CAPTURE = os.path.join(PERSISTENT_ROOT, "producer_performance.stdout")
+    STDERR_CAPTURE = os.path.join(PERSISTENT_ROOT, "producer_performance.stderr")
+    LOG_DIR = os.path.join(PERSISTENT_ROOT, "logs")
+    LOG_FILE = os.path.join(LOG_DIR, "producer_performance.log")
+    LOG4J_CONFIG = os.path.join(PERSISTENT_ROOT, "tools-log4j.properties")
+
     logs = {
+        "producer_performance_stdout": {
+            "path": STDOUT_CAPTURE,
+            "collect_default": True},
+        "producer_performance_stderr": {
+            "path": STDERR_CAPTURE,
+            "collect_default": True},
         "producer_performance_log": {
-            "path": "/mnt/producer-performance.log",
+            "path": LOG_FILE,
             "collect_default": True},
+        "jmx_log": {
+            "path": "/mnt/jmx_tool.log",
```

Comment:
After looking into the WARNINGs you mentioned collecting logs, it might make sense to set the jmx_log collect_default to True iff `jmx_object_names is not None`